pkg/event/base_publisher.go

```diff
@@ -54,3 +56,26 @@ func (p *publisher) Publish(data interface{}) {
 	msg, _ := json.Marshal(data)
 	_, _ = p.client.PublishAsync(string(p.subject), msg, p.ackHandler)
 }
+
+// PublishSync sends a message to the designated subject using a synchronous approach.
+func (p *publisher) PublishSync(data interface{}) error {
+	if os.Getenv("ENV") == "test" {
```

Comment:
Hi, Why do we need to check the environment here?

Reply:
I think it is an old setup that requires updates; otherwise, the unit tests will fail.

Reply:
Thank you for the clarification!